### ISSUE TITLE
Booleans in better-sqlite3

### ISSUE THREAD

**Issue by joshkel** [knex/knex#4955]
# Environment

Knex version: 1.0.1
Database + version: better-sqlite3 7.4.6
OS: macOS 12.1

# Bug

I'm trying out the new better-sqlite3 driver in Knex 1.x (due to the maintenance questions I saw with sqlite3 and [@vscode/sqlite3](https://github.com/microsoft/vscode-node-sqlite3/issues/14)). I ran into a problem within my application's code: Knex's better-sqlite3 driver [encodes booleans as string literals](https://github.com/knex/knex/pull/4871/files#diff-cde9b682a8cffbcbfda8552e00e2195defec3ef422a1c2083ecc923a3a92d5dfR58) (`'true`' and `'false'`), instead of as 0 and 1. This seems like a bug:

* It's incompatible with Knex's sqlite3 driver, so the better-sqlite3 driver can't be used as a drop-in replacement.
* It's inconsistent with [SQLIte's documentation](https://sqlite.org/quirks.html#no_separate_boolean_datatype).
* 0 and 1 are truthy and falsy in JavaScript, so they can often be used without further processing; both `'true'` and `'false'` are truthy.

```js
const knex = require('knex');

async function main() {
  for (const client of ['sqlite3', 'better-sqlite3']) {
    console.log('Testing ' + client);

    const db = knex({ client, connection: ':memory:', useNullAsDefault: true });
    await db.schema.createTable('test', (table) => {
      table.boolean('value').notNullable();
    });

    await db('test').insert([{ value: true }, { value: false }]);
    console.log(await db('test').select());

    await db.destroy();
    console.log();
  }
}

main();
```

**Comment by toBeOfUse** [comment]
(For future reference, this issue is fixed in PR #4982, it can now be closed at the maintainers' convenience I believe, thanks @joshkel for finding and linking to the exact line of code that had to be changed)

**Comment by kibertoad** [comment]
Released in 1.0.2
